server/routes/tasks.ts

```diff
@@ -99,137 +77,109 @@ router.post('/tasks', async (req: Request, res: Response, next: NextFunction) =>
                 },
             ],
         });
-        const date = helper.setDate(
-            req.body.date,
-            req.body.time,
-            userData.church.timezone,
-        );
+        const taskDate = setDate(date, time, church.timezone);
         const task = await db.Task.create({
-            date: new Date(date.toString()),
-            churchId: userData.church.id,
-            roleId: req.body.roleId,
+            date: new Date(taskDate.toString()),
+            churchId: church.id,
+            roleId,
             userId: parseInt(userId, 10),
         });
-        const millisTillDate = new Date(date.toString()).getTime() - now.getTime();
-        setTimeout(function () {
-            task.update({
-                status: 'archived',
-            });
-        }, millisTillDate);
-        res.status(201).send(task);
+        return res.status(201).send(task);
     } catch (err) {
-        if (err instanceof TokenExpiredError || err instanceof JsonWebTokenError) {
-            res.status(401).send({ message: 'Unauthorized' });
-        } else {
-            res.status(503).send({ message: 'Server error, try again later' });
-        }
         next(err);
+        return res.status(503).send({ message: 'Server error, try again later' });
     }
 });
 
-router.delete('/tasks/:taskId', async (req: Request, res: Response, next) => {
+router.delete('/tasks/:taskId', certify, async (req: Request, res: Response, next: NextFunction) => {
     try {
-        jwt.verify(req.headers.authorization, cert);
         const task = await db.Task.findOne({
             where: { id: req.params.taskId },
         });
-        if (task) {
-            await task.destroy().then(function () {
-                res.status(200).send({ message: 'Task deleted' });
-            });
-        } else {
-            res.status(404).send({ message: 'Task not found' });
-        }
+        const [message, status] = task ? ['Task deleted', 200] : ['Task not found', 404];
+        // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+        status === 200 && (await task.destroy());
```

Comment:
I think this should work too:
```suggestion
        if (status === 200) await task.destroy();
```

Reply:
yeah I think this is probably better